### PR TITLE
Update make scripts to point to correct docker image library

### PR DESCRIPTION
**Pull request by wilkermichael**
Changes proposed in this PR:
- Updated `make prepare-dev` and `make prepare-release` to set the correct docker images depending on if its in dev mode or release mode.

We expect dev to point to the docker images mirror library: `docker.mirror.hashicorp.services/hashicorppreview/consul-k8s-control-plane:<consul-k8s-version-dev>`
We expect release to point to the live docker images library: `hashicorp/consul-k8s-control-plane:<consul-k8s-version>`

How I've tested this PR:
- Ran `make prepare-dev` and `make prepare-release` locally

How I expect reviewers to test this PR:
- Look at the example commits I've put up

Checklist:
- [N/A] Tests added
- [N/A] CHANGELOG entry added 
  > HashiCorp engineers only, community PRs should not add a changelog entry.
  > Entries should use present tense (e.g. Add support for...)

### CODE REVIEW

**Review by curtbushko — APPROVED** [review]
Glorious!

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,7 @@
 ## UNRELEASED
 
+## 1.1.0-beta1 (October 4, 2022)
+
```
Comment (kschoche) [inline CHANGELOG.md:4]:
```suggestion
```

Comment (wilkermichael) [reply CHANGELOG.md:4]: Yeah, that's a byproduct of running the script multiple times, this wouldn't be a nominal scenario.

I'm going to drop the example commits before completing the PR

**Review by kschoche — APPROVED** [review]
🔥

**Pull request merged**